File: service-core/src/main/java/com/timothy/service_core/ServiceComponent.java
package com.timothy.service_core;

import com.timothy.spotifyarchitecture.room.SpotifyModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by tim on 5/31/17.
 */
@Singleton
@Component(modules = ServiceModule.class)
public interface ServiceComponent {
    //void inject(MainActivity mainActivity);
}

File: app/src/main/java/com/timothy/musicall/ServiceComponent.kt
package com.timothy.musicall

import dagger.Component
import javax.inject.Singleton

/**
 * Creates a Dagger 2.0 component interface to inject MainActivity
 */
@Singleton
@Component(modules = arrayOf(ServiceModule::class))
interface ServiceComponent {
    fun inject(mainActivity: MainActivity)
}
